app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,57 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        :gce => [ManageIQ::Providers::Google::CloudManager, :name],
+        :aws => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      instance_uri = URI(provider_id)
+      prov, name_field = providers[instance_uri.scheme.to_sym]
+      return nil unless prov
+      project_name = instance_uri.host
+      _, az, instance_name = instance_uri.path.split('/')
+
+      instance_details = {name_field => instance_name}
+      instance_details[:availability_zone_id] =
+        AvailabilityZone.find_by(:name => az).try(:id) if az != ""
+
+      if project_name
+        project = prov.find_by_project(project_name)
+        return nil unless project
+        prov = project.vms
```

Comment:
With GCE I understand that they are unique [1] but with AWS IIRC then two providers might have two different instances, one in each,  with the same instance-id. they might even be in the same availability-zone so this wouldn't help anyway. [2][3]

[1] https://cloud.google.com/compute/docs/reference/latest/instances
[2] http://serverfault.com/questions/58401/is-the-amazon-ec2-instance-id-unique-forever
[3] http://www.jackofallclouds.com/2009/09/anatomy-of-an-amazon-ec2-resource-id/